oak-search/src/test/java/org/apache/jackrabbit/oak/plugins/index/ExcerptTest.java

```diff
@@ -0,0 +1,340 @@
+                assertNotNull("rep:excerpt not evaluated", nodeExcerpt);
+                excerpt1 = nodeExcerpt.getValue(STRING);
+                assertTrue("rep:excerpt didn't evaluate correctly - got '" + excerpt1 + "'",
+                        "is <strong>fox</strong> ifoxing".equals(excerpt1) || "ifoxing <strong>fox</strong>".equals(excerpt1));
+
+                nodeExcerpt = firstRow.getValue("rep:excerpt(.)");
+                assertNotNull("rep:excerpt(.) not evaluated", nodeExcerpt);
+                excerpt2 = nodeExcerpt.getValue(STRING);
+                assertEquals("excerpt extracted via rep:excerpt not same as rep:excerpt(.)", excerpt1, excerpt2);
+            } catch (ParseException e) {
+                fail(e.getMessage());
+            }
+        });
+    }
+
+    @Test
+    public void nonIndexedRequestedPropExcerpt() throws Exception {
+        Tree contentRoot = root.getTree("/").addChild("testRoot");
+        contentRoot.setProperty("foo", "fox");
+        contentRoot.setProperty("baz", "is fox ifoxing");
+        root.commit();
+
+        assertEventually(() -> {
+            String query = "SELECT [rep:excerpt(baz)] FROM [nt:base] WHERE CONTAINS(*, 'fox')";
+            Result result = null;
+            try {
+                result = executeQuery(query, SQL2, NO_BINDINGS);
+                Iterator<? extends ResultRow> resultIter = result.getRows().iterator();
+                assertTrue(resultIter.hasNext());
+                ResultRow firstRow = resultIter.next();
+
+                PropertyValue nodeExcerpt = firstRow.getValue("rep:excerpt(baz)");
+                assertNull("rep:excerpt(baz) if requested explicitly must be indexed to be evaluated", nodeExcerpt);
+            } catch (ParseException e) {
+                fail(e.getMessage());
+            }
+        });
+    }
+
+    @Test
+    public void propExcerpt() throws Exception {
+        Tree contentRoot = root.getTree("/").addChild("testRoot");
+        contentRoot.setProperty("foo", "is fox ifoxing");
+        root.commit();
+
+        String query = "SELECT [rep:excerpt(foo)] FROM [nt:base] WHERE CONTAINS(*, 'fox')";
+        assertEventually(() -> {
+            Result result = null;
+            try {
+                result = executeQuery(query, SQL2, NO_BINDINGS);
+                Iterator<? extends ResultRow> resultIter = result.getRows().iterator();
+                assertTrue(resultIter.hasNext());
+                ResultRow firstRow = resultIter.next();
+
+                PropertyValue nodeExcerpt;
+                String excerpt;
+
+                nodeExcerpt = firstRow.getValue("rep:excerpt(foo)");
+                assertNotNull("rep:excerpt(foo) not evaluated", nodeExcerpt);
+                excerpt = nodeExcerpt.getValue(STRING);
+                assertTrue("rep:excerpt(foo) didn't evaluate correctly - got '" + excerpt + "'",
+                        "is <strong>fox</strong> ifoxing".equals(excerpt));
```

Comment:
this can be simplified 

```suggestion
                assertEquals("rep:excerpt(foo) didn't evaluate correctly - got '" + excerpt + "'", 
                        "is <strong>fox</strong> ifoxing", excerpt);
```

Reply:
This was the original test code.